controllers/discovery/initDiscoveryUserData.js

```diff
@@ -46,6 +48,7 @@ const InitDiscoveryUserData = async (req, res) => {
         ON A.user_id = B.user_id_followed AND B.user_id_follower = :userId
         WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false
         ORDER BY
+        COALESCE(A.karma_score, 0) DESC,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change is good, but it might be better to handle the case where `karma_score` is null in a more explicit way. Currently, if `karma_score` is null, it will be replaced with 0 which might not be the intended behavior. If you want to exclude users with null `karma_score`, you should add this condition to the WHERE clause.
```diff
-        COALESCE(A.karma_score, 0) DESC,
+        A.karma_score DESC,
```
And in the WHERE clause:
```diff
-        WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false
+        WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false AND A.karma_score IS NOT NULL
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->